### PR TITLE
More Modcomp shit

### PR DESCRIPTION
**Pull request by aidan-bowman**
## About the Pull Request

Adds in a basic upload server system, which I will use in the future for an overhaul of the Archivist area. Also deletes some useless programs (e.g. newscaster) and assets (e.g. old PDA images). General code cleanup in the modcomp system.

## Why It's Good For The Game

In the short term, I can see 2 main uses for this basic database system.
1. Moving all of the physical paper documents on SCPs to a server room, saving on resources as well as allowing for more interesting gameplay.
2. Moving downloadable programs from a transient online service to a server room, allowing for sabotage and better control over file access.

## Future Improvements

Obviously, the upload database should be given more use in the future.
I didn't make any sprites for the two new programs.

## Changelog

:cl:
add: Adds new database server and client programs, used to host a basic file browser service.
del: Removed newscaster program and flight program reports
imagedel: Deleted newspaper images and old PDA images.
spellcheck: Changed some baystation references to SCP references.
code: General code cleanup in modcomp files as well as some helper procs.
/:cl:

### CODE REVIEW

**Comment by TheDarkElites** [comment]
Omg no way hes gonna finish mod comps stuff!!!

We stan cheese!!!

File: code/modules/modular_computers/file_system/programs/servers/upload_database.dm
Comment (TheDarkElites) [inline code/modules/modular_computers/file_system/programs/servers/upload_database.dm]: Use the auto doc feature for these.
```
/// comment
var/whatever_the_fuck
```

File: code/modules/modular_computers/file_system/programs/servers/upload_database.dm
Comment (TheDarkElites) [inline code/modules/modular_computers/file_system/programs/servers/upload_database.dm]: ???

File: code/modules/modular_computers/file_system/programs/servers/upload_database.dm
Comment (TheDarkElites) [inline code/modules/modular_computers/file_system/programs/servers/upload_database.dm]: more stuff you left commented out

File: code/modules/modular_computers/file_system/programs/servers/upload_database_c.dm
Comment (TheDarkElites) [inline code/modules/modular_computers/file_system/programs/servers/upload_database_c.dm]: Use auto doc here too

File: code/modules/modular_computers/file_system/programs/servers/upload_database_c.dm
Comment (TheDarkElites) [inline code/modules/modular_computers/file_system/programs/servers/upload_database_c.dm]: Auto doc for the proc defines aswell

**Review by TheDarkElites — CHANGES_REQUESTED** [review]
Stuff you should fix
Also, you should get Tusurpta to look at this after you fix the things I mentioned since he's a much better maintainer than me.

**Comment by TheDarkElites** [comment]
This looks good enough for me, will assign it as waiting for test merge.

**Comment by TheDarkElites** [comment]
[17:06:33] Runtime in virus.dm, line 154: Cannot read null.access
proc name: process tick (/datum/computer_file/program/virus/process_tick)
src: /datum/computer_file/program/v... (/datum/computer_file/program/virus)
call stack:
/datum/computer_file/program/v... (/datum/computer_file/program/virus): process tick()
the laptop computer (/obj/item/modular_computer/laptop/preset/custom_loadout/advanced): Process(10, 4953, Objs (/datum/controller/subsystem/processing/obj))
Objs (/datum/controller/subsystem/processing/obj): fire(0)
Objs (/datum/controller/subsystem/processing/obj): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop()
Master (/datum/controller/master): StartProcessing(0)

File: code/game/movietitles.dm
Comment (TheDarkElites) [inline code/game/movietitles.dm:127]: get rid of the global GetAssignment proc all together if you're gonna replace all of the use cases.

**Review by TheDarkElites — CHANGES_REQUESTED** [review]

File: code/game/movietitles.dm
Comment (TheDarkElites) [reply code/game/movietitles.dm:127]: nevermind im fucking blind apparently

File: code/modules/modular_computers/file_system/reports/crew_record.dm
Comment (TheDarkElites) [inline code/modules/modular_computers/file_system/reports/crew_record.dm:149]: move above //Global methods comment since this is now a local proc

**Review by TheDarkElites — CHANGES_REQUESTED** [review]

File: code/modules/modular_computers/file_system/reports/crew_record.dm
Comment (aidan-bowman) [reply code/modules/modular_computers/file_system/reports/crew_record.dm:149]: done (nerd)

**Review by TheDarkElites — APPROVED** [review]
Tsurpta said this is good, it looks good to me, and it passed TM

god bless the modcomp pr

**Pull request merged**
